suite-native/coin-enabling/src/components/CoinEnablingForm.tsx

```diff
@@ -0,0 +1,94 @@
+import { useDispatch, useSelector } from 'react-redux';
+
+import { useNavigation } from '@react-navigation/native';
+
+import { changeCoinVisibility } from '@suite-common/wallet-core';
+import { useAlert } from '@suite-native/alerts';
+import { EventType, analytics } from '@suite-native/analytics';
+import {
+    selectDeviceEnabledDiscoveryNetworkSymbols,
+    selectDiscoveryNetworkSymbols,
+} from '@suite-native/discovery';
+import { Form, useForm } from '@suite-native/forms';
+import { Translation } from '@suite-native/intl';
+
+import { DiscoveryCoinsFilter } from './DiscoveryCoinsFilter';
+import { CoinEnablingFormValues, coinEnablingFormValidationSchema } from '../coinEnablingSchema';
+
+export const CoinEnablingForm = () => {
+    const dispatch = useDispatch();
+    const navigation = useNavigation();
+
+    const enabledNetworkSymbols = useSelector(selectDeviceEnabledDiscoveryNetworkSymbols);
+    const networkSymbols = useSelector(selectDiscoveryNetworkSymbols);
+
+    const { showAlert } = useAlert();
+
+    const showLastNetworkAlert = useCallback(
+        () =>
+            showAlert({
+                title: <Translation id="moduleSettings.coinEnabling.oneNetworkSymbolAlert.title" />,
+                description: (
+                    <Translation id="moduleSettings.coinEnabling.oneNetworkSymbolAlert.description" />
+                ),
+                primaryButtonTitle: <Translation id="generic.buttons.gotIt" />,
+                primaryButtonVariant: 'redBold',
+            }),
+        [showAlert],
+    );
+
+    const form = useForm<CoinEnablingFormValues>({
+        defaultValues: {
+            enabledCoins: enabledNetworkSymbols,
+        },
+        validation: coinEnablingFormValidationSchema,
+    });
+
+    const handleSubmit = form.handleSubmit(values => {
+        const changedCoins = networkSymbols.filter(
+            symbol =>
+                enabledNetworkSymbols.includes(symbol) !== values.enabledCoins.includes(symbol),
+        );
+
+        if (changedCoins.length === 0) return;
+
+        changedCoins.forEach(symbol => {
+            const isEnabled = values.enabledCoins.includes(symbol);
+            dispatch(changeCoinVisibility({ symbol, shouldBeVisible: isEnabled }));
+
+            analytics.report({
+                type: EventType.SettingsChangeCoinEnabled,
+                payload: {
+                    symbol,
+                    value: isEnabled,
+                },
+            });
+        });
+    });
+
+    useEffect(
+        () =>
+            navigation.addListener('beforeRemove', () => {
+                if (form.formState.isDirty) {
+                    handleSubmit();
+                }
+            }),
+        [navigation, form.formState.isDirty, handleSubmit],
+    );
+
+    const filterProps = useMemo<ComponentProps<typeof DiscoveryCoinsFilter>>(
+        () => ({
+            allowDeselectLastCoin: false,
```

Comment:
This is unused if I'm not overlooking something. IMO, this const is not necessary and you could use a selector in `DiscoveryCoinsFilter.tsx` and only send `onDisablingLastCoin` as an optional prop.

Reply:
Or if its necessary due to rerendering, it's not super obvious from the code so maybe a comment might help?

Reply:
No, the whole filterProps const is unnecesary complexity, thanks for pointing that out. But the callback makes sense to me since it's passed to child component.

1ec9d13a61c3f7cea651445e9b7158993ee47b4e